CleTimer/CleTimer/default/model.py

```diff
@@ -0,0 +1,173 @@
+        soi = x
+
+        for i in range(len(soi)):
+            if len(soi[i]) < 80:
+                soi[i] = elongate_intron(soi[i])
+
+        parameters_batch = self._construct_features_array(soi)
+
+        don_cleavage_time = self.don_model.predict(parameters_batch)
+        acc_cleavage_time = self.acc_model.predict(parameters_batch)
+
+        cleavage_time = {'acc_cleavage_time': np.exp(acc_cleavage_time),
+                         'don_cleavage_time': np.exp(don_cleavage_time)}
+
+        return cleavage_time
+
+    def _construct_features_array(self, soi):
+        """
+        Constructs features array.
+
+        :return: numpy array for running the model.
+        """
+
+        shape = (len(soi), len(self.features_metadata))
+        batch_encoded_features = np.zeros(shape)
+
+        # first feature is the gc content in acceptor region (double acceptor window at the end)
+        acceptors = [x[2 * self.acc_i:] for x in soi]
+        batch_encoded_features[:, 0] = np.array([self._count_gc_content(acceptor) for acceptor in acceptors])
+        # second feature is gc content in intron region
+        introns = [x[self.don_i: self.acc_i] for x in soi]
+        batch_encoded_features[:, 1] = np.array([self._count_gc_content(intron) for intron in introns])
+
+        # get the list of bp index for each sequence of batch
+        self.bp_indexes = self._get_bp_indexes_labranchor(soi)
+        # slice out feature sequences
+        seqA = [seq[self.acc_i - 4: self.acc_i + 6] for seq in soi]
+        seqB = [soi[j][int(self.bp_indexes[j]) - 15: int(self.bp_indexes[j]) + 6] for j in range(len(soi))]
+        seqD = [seq[self.don_i - 3: self.acc_i + 16] for seq in soi]
+
+        # fill out the rest of the features (base-by-region features)
+        for i in range(2, len(self.features_metadata)):
+            # parse the current feature info
+            (region, pos, nucl) = self.features_metadata[i]
+            if (region == 'seqD' or region == 'seqA') and pos > 0:  # decrement, since acc_i/don_i is pos = 1
+                pos -= 1
+
+        #    for j in range( len(soi) ):
+        #        if region == 'seqB':
+        #            i_oi = int(self.bp_indexes[j]) + int(pos)
+        #            if soi[j][i_oi].upper() == nucl:
+        #                batch_encoded_features[j, i] = 1
+        #        else:
+        #            if region == 'seqA' and soi[j][ (self.acc_i + int(pos)) ].upper() == nucl:
+        #                batch_encoded_features[j, i] = 1
+        #            elif region == 'seqD' and soi[j][ (self.don_i + int(pos)) ].upper() == nucl:
+        #                batch_encoded_features[j, i] = 1
+       #   executor = concurrent.futures.ProcessPoolExecutor(10)
+       #    futures = [executor.submit(work_seq_on_feature, seqA[j], seqB[j], seqD[j], region, pos, nucl, j) for j in range(len(soi))]
+       #    concurrent.futures.wait(futures)
+
+            pool = ProcessPool(nodes=10)
```

Comment:
@stefanches7 I multiprocessing really required here? Seem shat the function `work_seq_on_feature` could be speedup by using numpy or pandas.

Reply:
also. I would rather re-use a pool of workers across the batches and also the above for loop rather than start a new processPool for every feature

Reply:
@Avsecz no, it's not really required. I only want to speed-up the general process of parsing features. And the problem with the multiprocessing is pickling the module, as said, so the first thing I wanted to inquire about is how to do it properly in the context of kipoi.

Reply:
But if I can speed up without multiprocessing, I would 👍 how can I speed up the function with pandas? I guess the most workload comes from indexing the string and extracting features, so I would maybe try to use a more efficient datatype to do it.

Reply:
So yes, that's basically the main point of the PR @Avsecz . Feature extraction as it is now runs really slow and needs to be speeded up. I think about this or that way to do it faster taking n account the framework; if you have some ideas I'm glad to hear them.